r/tests/testthat/test-dplyr-summarize.R

```diff
@@ -459,6 +482,35 @@ test_that("quantile()", {
   )
 })
 
+test_that("quantile() with namespacing", {
+  suppressWarnings(
+    expect_warning(
+      expect_equal(
+        tbl %>%
+          group_by(some_grouping) %>%
+          summarize(
+            q_dbl = quantile(dbl, probs = 0.5, na.rm = TRUE, names = FALSE),
+            q_int = as.double(
+              quantile(int, probs = 0.5, na.rm = TRUE, names = FALSE)
+            )
+          ) %>%
+          arrange(some_grouping),
+        Table$create(tbl) %>%
+          group_by(some_grouping) %>%
+          summarize(
+            q_dbl = stats::quantile(dbl, probs = 0.5, na.rm = TRUE),
```

Comment:
Now this is a test you should do with separate namespaced and not versions so you can test that both raise the warning.

Reply:
Yep, that's how I did it: [namespaced version](https://github.com/dragosmg/arrow/blob/642c23ee865140bdbdee7e416558aecc580f8500/r/tests/testthat/test-dplyr-summarize.R#L485-L512) and [non-namespaced version](https://github.com/dragosmg/arrow/blob/642c23ee865140bdbdee7e416558aecc580f8500/r/tests/testthat/test-dplyr-summarize.R#L404-L429).

Reply:
My point was that because you're calling both namespaced and non-namespaced versions in the same call, you're not testing that both versions raise the warning: you expect one warning, which could come from any of them, and suppress the rest of the warnings. But I looked again and the warning is coming from the `quantile` registered function, so as long as you're getting the same numbers either way you call it, you're passing through the same code that triggers the warnings. So it's ok.